apps/shell/tash_main.c

```diff
@@ -149,14 +153,17 @@ char *tash_read_input_line(int fd)
 		FD_SET(fd, &tfd);
 
 		if ((select(fd + 1, &tfd, NULL, NULL, &stimeout)) && FD_ISSET(fd, &tfd)) {
+#endif
+#ifdef CONFIG_PM
+			/* Ensure board does not go to sleep for TASH_PM_TIMEDSUSPEND_TIME_IN_MS*/
+			tash_pm_timedsuspend(TASH_PM_TIMEDSUSPEND_TIME_IN_MS);
```

Comment:
Placement of this looks a bit weird, I think we are expected to wait for N seconds before going to sleep after the user has entered a character, right? Here we are suspending sleep for N seconds even when user has not entered anything. Shouldnt this be below the read function?

```
			nbytes = read(fd, &buffer[pos], (bufsize - pos));
			if (nbytes < 0) {
				shdbg("TASH: can not read uart\n");
				return buffer;
			}
                        /* Ensure board does not go to sleep for TASH_PM_TIMEDSUSPEND_TIME_IN_MS as user started entering input*/
		------->tash_pm_timedsuspend(TASH_PM_TIMEDSUSPEND_TIME_IN_MS);
```

Reply:
Done